File: recibidor.py
import pika

# Establecer la conexión con el servidor RabbitMQ
connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
channel = connection.channel()

# Declarar una cola llamada 'cola_ruby_mensajes'
channel.queue_declare(queue='cola_ruby_mensajes')

print(' [*] Esperando mensajes. Para salir, presione CTRL+C')

# Callback para procesar los mensajes recibidos
def callback(ch, method, properties, body):
    print(" [x] Recibido '{}' desde Python".format(body.decode()))

# Suscribirse a la cola y procesar los mensajes recibidos
channel.basic_consume(queue='cola_ruby_mensajes', on_message_callback=callback, auto_ack=True)

# Comenzar a escuchar los mensajes
channel.start_consuming()

File: recibidor.rb
require 'bunny'

# Conexión con el servidor RabbitMQ
connection = Bunny.new(hostname: "localhost")
connection.start

# Abre un canal
channel = connection.create_channel

# Se declara una cola llamada 'cola_mensajes'
queue = channel.queue('cola_mensajes')

puts ' [*] Esperando mensajes. Para salir, presione CTRL+C'

begin
  # Suscribirse a la cola y procesar los mensajes recibidos
  queue.subscribe(block: true) do |delivery_info, properties, body|
    puts " [x] Recibido '#{body}' desde Ruby"
  end
rescue Interrupt => _
  connection.close
  exit(0)
end
